cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -147,151 +136,80 @@ func (gc *GceCache) GetMigs() []Mig {
-	return migRefs
-}
-
-// Methods locking on cacheMutex.
-
-// GetMigForInstance returns Mig to which the given instance belongs.
-// Attempts to regenerate cache if there is a Mig with matching prefix in migs list.
-// TODO(aleksandra-malinowska): reconsider failing when there's a Mig with
-// matching prefix, but instance doesn't belong to it.
-func (gc *GceCache) GetMigForInstance(instanceRef GceRef) (Mig, error) {
+// GetMigForInstance returns the cached MIG for instance GceRef
+func (gc *GceCache) GetMigForInstance(instanceRef GceRef) (GceRef, bool) {
 	gc.cacheMutex.Lock()
 	defer gc.cacheMutex.Unlock()
 
-	if migRef, found := gc.instanceRefToMigRef[instanceRef]; found {
-		mig, found := gc.getMigNoLock(migRef)
-		if !found {
-			return nil, fmt.Errorf("instance %+v belongs to unregistered mig %+v", instanceRef, migRef)
-		}
-		return mig, nil
-	} else if _, found := gc.instancesFromUnknownMigs[instanceRef]; found {
-		return nil, nil
-	}
-
-	for _, migRef := range gc.getMigRefs() {
-
-		// get mig basename - refresh if not found
-		// todo[lukaszos] move this one as well as whole instance cache regeneration out of cache
-		migBasename, found := gc.migBaseNameCache[migRef]
-		var err error
-		if !found {
-			migBasename, err = gc.GceService.FetchMigBasename(migRef)
-			if err != nil {
-				return nil, err
-			}
-			gc.migBaseNameCache[migRef] = migBasename
-		}
-
-		if migRef.Project == instanceRef.Project &&
-			migRef.Zone == instanceRef.Zone &&
-			strings.HasPrefix(instanceRef.Name, migBasename) {
-			if err := gc.regenerateInstanceCacheForMigNoLock(migRef); err != nil {
-				return nil, fmt.Errorf("error while looking for MIG for instance %+v, error: %v", instanceRef, err)
-			}
-
-			migRef, found := gc.instanceRefToMigRef[instanceRef]
-			if !found {
-				klog.Warningf("instance %+v belongs to unknown mig", instanceRef)
-				gc.instancesFromUnknownMigs[instanceRef] = struct{}{}
-				return nil, nil
-			}
-			mig, found := gc.getMigNoLock(migRef)
-			if !found {
-				return nil, fmt.Errorf("instance %+v belongs to unregistered mig %+v", instanceRef, migRef)
-			}
-			return mig, nil
-		}
-	}
-	// Instance doesn't belong to any configured mig.
-	return nil, nil
-}
-
-func (gc *GceCache) removeInstancesForMigs(migRef GceRef) {
-	for instanceRef, instanceMigRef := range gc.instanceRefToMigRef {
-		if migRef == instanceMigRef {
-			delete(gc.instanceRefToMigRef, instanceRef)
-			delete(gc.instancesFromUnknownMigs, instanceRef)
-		}
+	migRef, found := gc.instancesToMig[instanceRef]
+	if found {
+		klog.V(5).Infof("MIG cache hit for %s", instanceRef)
 	}
+	return migRef, found
 }
 
-func (gc *GceCache) getMigNoLock(migRef GceRef) (mig Mig, found bool) {
-	mig, found = gc.migs[migRef]
-	return
-}
-
-// RegenerateInstanceCacheForMig triggers instances cache regeneration for single MIG under lock.
-func (gc *GceCache) RegenerateInstanceCacheForMig(migRef GceRef) error {
-	klog.V(4).Infof("Regenerating MIG information for %s", migRef.String())
```

Comment:
Do we think those were useless? I'm not saying they weren't, just want to make sure this is a conscious decision.

Reply:
I've added the line back, did not mean to remove it in the first place.